CMakeLists.txt

```diff
@@ -1,42 +1,45 @@
-cmake_minimum_required(VERSION 3.0...3.16)
-project(OPENNS LANGUAGES C CXX VERSION 2.0.0)
+cmake_minimum_required(VERSION 3.15)
```

Comment:
Why are you forcing 3.15 as minimum CMake version? What do we specifically need from 3.15?

Reply:
Here's the things we are using from various CMake versions.

3.1
    CXX_STANDARD

3.7
    "Visual Studio 15 2017" generator

3.8
    UseSWIG module gained swig_add_library()

3.12
    new FindPython3 module
    UseSWIG updates

3.13
    added cmake -S .
    added cmake -B build

3.15
    CMP0092 (remove /W3 from default MSVC compiler flags)

Reply:
I have concerns about this approach to pushing the minimum CMake to a version released only last July. Specially, it is for a minor feature: https://gitlab.kitware.com/cmake/cmake/commit/1baf122cd41d6500139649661052236b8e684565